ocaml/runtime/extern.c

```diff
@@ -1279,8 +1240,26 @@ intnat reachable_words_once(value root, enum reachable_words_traversal_mode mode
         } else {
           extern_update_location_with_data(h, new_mark);
         }
+
         /* The block contributes to the total size */
         size += 1 + sz;           /* header word included */
+        if (sizes_by_root_id) {
```

Comment:
(after some offline discussion with Stephen) Since this is a `value`, we should really be using a proper `value` as the sentinel. So this should actually be `if (sizes_by_root_id == Val_unit)` and the corresponding call site should use `Val_unit` rather than 0.

Reply:
Makes sense